packages/aragon-wrapper/src/index.js

```diff
@@ -457,6 +475,276 @@ export default class Aragon {
+        const update$ = of(...updatedRepoAppIds).pipe(
+          map((appId) => ({ appId }))
+        )
+
+        // Create a new observable to project each new repo as its own emission.
+        const newRepo$ = of(...newRepos)
+
+        // Create a new observable to project each new repo's address as its own update emission.
+        const repoAddress$ = newRepo$.pipe(
+          map(({ appId, repoProxy }) => ({
+            appId,
+            repoAddress: repoProxy.address
+          }))
+        )
+
+        // Create a new observable that projects each NewVersion event as its own update event
+        // emission.
+        // This one is a bit trickier, due to the higher order observable. Keep reading.
+        const version$ = newRepo$.pipe(
+          // `mergeMap()` to "flatten" the async transformation. This async function returns an
+          // observable, which is ultimately the NewVersion stream. More on this, after the break.
+          // Note: we don't care about the ordering, so we use `mergeMap()` instead of `concatMap()`
+          mergeMap(async ({ appId, repoProxy }) => {
+            const initialVersions = [
+              // Immediately query state from the repo contract, to avoid having to wait until all
+              // past events sync (may be long)
+              ...await getAllRepoVersions(repoProxy)
+            ]
+
+            // Return an observable subscribed to NewVersion events, giving us:
+            //   - Timestamps for versions that were published prior to this process running
+            //   - Notifications for newly published versions
+            //
+            // Reduce this with the cached version information to emit version updates for the repo.
+            return repoProxy.events('NewVersion').pipe(
+              // Project each event to a new version info object, one at a time
+              concatMap(async (event) => {
+                const { versionId: eventVersionId } = event.returnValues
+
+                // Adjust from Ethereum time
+                const timestamp = (await this.web3.eth.getBlock(event.blockNumber)).timestamp * 1000
+
+                const versionIndex = initialVersions.findIndex(({ versionId }) => versionId === eventVersionId)
+                const versionInfo =
+                  versionIndex === -1
+                    ? await getRepoVersionById(repoProxy, eventVersionId)
+                    : initialVersions[versionIndex]
+
+                return {
+                  ...versionInfo,
+                  timestamp
+                }
+              }),
+
+              // Trick to immediately emit (e.g. similar to a do/while loop)
+              startWith(null),
+
+              // Reduce newly emitted versions into the full list of versions
+              scan(({ appId, versions }, newVersionInfo) => {
+                let newVersions = versions
+                if (newVersionInfo) {
```

Comment:
is `newVersionInfo` ever falsey ?

Reply:
Directly preceding it, with the [`startsWith(null)`](https://github.com/aragon/aragon.js/pull/268/files/e02f0f1f5092fab2cf557dae1d6e920f0423d5c6#diff-5b914b5b526a57f89974fa2fe009ee80R620).

Reply:
is the [`startsWith(null)`](https://github.com/aragon/aragon.js/pull/268/files/e02f0f1f5092fab2cf557dae1d6e920f0423d5c6#diff-5b914b5b526a57f89974fa2fe009ee80R620) to flush the pipeline with whatever we already have in `initialVersions`?  In

Reply:
Yup, exactly, so we don't need to wait for the `eth_getlogs` to finish